crates/bevy_feathers/src/controls/toggle_switch.rs

```diff
@@ -0,0 +1,249 @@
+}
+
+/// Marker for the toggle switch outline
+#[derive(Component, Default, Clone)]
+struct ToggleSwitchOutline;
+
+/// Marker for the toggle switch slide
+#[derive(Component, Default, Clone)]
+struct ToggleSwitchSlide;
+
+/// Template function to spawn a toggle switch.
+///
+/// # Arguments
+/// * `props` - construction properties for the toggle switch.
+/// * `overrides` - a bundle of components that are merged in with the normal toggle switch components.
+pub fn toggle_switch<B: Bundle>(props: ToggleSwitchProps, overrides: B) -> impl Bundle {
+    (
+        Node {
+            width: size::TOGGLE_WIDTH,
+            height: size::TOGGLE_HEIGHT,
+            border: UiRect::all(Val::Px(2.0)),
+            ..Default::default()
+        },
+        CoreCheckbox {
+            on_change: props.on_change,
+        },
+        ToggleSwitchOutline,
+        BorderRadius::all(Val::Px(5.0)),
+        ThemeBackgroundColor(tokens::SWITCH_BG),
+        ThemeBorderColor(tokens::SWITCH_BORDER),
+        AccessibilityNode(accesskit::Node::new(Role::Switch)),
+        Hovered::default(),
+        CursorIcon::System(bevy_window::SystemCursorIcon::Pointer),
+        TabIndex(0),
+        overrides,
+        children![(
+            Node {
+                position_type: PositionType::Absolute,
+                left: Val::Percent(0.),
+                top: Val::Px(0.),
+                bottom: Val::Px(0.),
+                width: Val::Percent(50.),
+                ..Default::default()
+            },
+            BorderRadius::all(Val::Px(3.0)),
+            ToggleSwitchSlide,
+            ThemeBackgroundColor(tokens::SWITCH_SLIDE),
+        )],
+    )
+}
+
+fn update_switch_styles(
+    q_switches: Query<
+        (
+            Entity,
+            Has<InteractionDisabled>,
+            Has<Checked>,
+            &Hovered,
+            &ThemeBackgroundColor,
+            &ThemeBorderColor,
+        ),
+        (
+            With<ToggleSwitchOutline>,
+            Or<(Changed<Hovered>, Added<Checked>, Added<InteractionDisabled>)>,
+        ),
+    >,
+    q_children: Query<&Children>,
+    mut q_slide: Query<(&mut Node, &ThemeBackgroundColor), With<ToggleSwitchSlide>>,
+    mut commands: Commands,
+) {
+    for (switch_ent, disabled, checked, hovered, outline_bg, outline_border) in q_switches.iter() {
+        let Some(slide_ent) = q_children
+            .iter_descendants(switch_ent)
+            .find(|en| q_slide.contains(*en))
+        else {
+            continue;
+        };
+        // Safety: since we just checked the query, should always work.
+        let (ref mut slide_style, slide_color) = q_slide.get_mut(slide_ent).unwrap();
+        set_switch_colors(
+            switch_ent,
+            slide_ent,
+            disabled,
+            checked,
+            hovered.0,
+            outline_bg,
+            outline_border,
+            slide_style,
+            slide_color,
+            &mut commands,
+        );
```

Comment:
You can call `set_switch_colors` inside the find block. The true false thing is silly, there should be a better construction but I can't think though need to go to sleep:
```suggestion
        q_children.iter_descendants(switch_ent).find(|&en| {
            if let Ok((ref mut slide_style, slide_color)) = q_slide.get_mut(en) {
                set_switch_colors(
                    switch_ent,
                    en,
                    disabled,
                    checked,
                    hovered.0,
                    outline_bg,
                    outline_border,
                    slide_style,
                    slide_color,
                    &mut commands,
                );
                true
            } else {
                false
            }
        });
```